sonar-batch/src/main/java/org/sonar/batch/scan/ProjectLock.java

```diff
@@ -55,13 +56,25 @@ public void tryLock() {
       if (lockFile == null) {
         failAlreadyInProgress(null);
       }
+      hideLockFileWindows(lockFilePath);
     } catch (OverlappingFileLockException e) {
       failAlreadyInProgress(e);
     } catch (IOException e) {
       throw new IllegalStateException("Failed to create project lock in " + lockFilePath.toString(), e);
     }
   }
-  
+
+  private static void hideLockFileWindows(Path p) {
+    try {
+      DosFileAttributeView fileAttrView = Files.getFileAttributeView(p, DosFileAttributeView.class);
+      if (fileAttrView != null) {
+        fileAttrView.setHidden(true);
+      }
+    } catch (IOException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)